File: ASC-RailwayStationWP-server/src/main/java/api/service/train/TrainService.java
package api.service.train;

import api.exception.TrainException;
import api.model.TrainBean;

import java.util.List;

/**
 * Created by nolesuk on 07-Mar-17.
 */
public interface TrainService {

    void addTrain();

    void addTrainRoute(TrainBean trainBean) throws TrainException;

    List<TrainBean> getTrainsByRoute(String arrivalStation, String departureStation);

    List<TrainBean> getTrainsByParams(String arrivalStation, String departureStation, String arrivalDate, String departureDate);

}

File: src/app/common/service/train.service.ts
import {AbstractHttpService} from './abstract-http.service';
import {Train} from '../model/train';
import {URLSearchParams} from '@angular/http';

export class TrainService extends AbstractHttpService {

  private getByParams = 'http://localhost:8080/getTrainsByParams';
  private addTrain = 'http://localhost:8080/addTrains';

  public add(train: Train): Promise<void> {
    return super.doPost(this.addTrain, train, null);
  }

  public searchRoutes(departureDate: Date, departureStation: string, arrivalDate: Date, arrivalStation: string): Promise<Train[]> {
    const options = this.getRouteOptions(departureStation, arrivalStation, departureDate, arrivalDate);
    return super.doPost(this.getByParams, {}, options).then(response => response.json() as Train[]);
  }

  private getRouteOptions(departureStation: string, arrivalStation: string, departureDate: Date, arrivalDate: Date) {
    const params = new URLSearchParams();
    params.set('departureStation', departureStation);
    params.set('arrivalStation', arrivalStation);
    params.set('departureDate', JSON.stringify(departureDate.getTime()));
    params.set('arrivalDate', JSON.stringify(arrivalDate.getTime()));
    const options = super.getRequestOptions();
    options.params = params;
    return options;
  }


}
